art/estimators/classification/pytorch.py

```diff
@@ -395,6 +404,58 @@ def loss_gradient(self, x, y, **kwargs):
 
         return grads
 
+    def loss_gradient_framework(self, x, y, **kwargs):
+        """
+        Compute the gradient of the loss function w.r.t. `x`.
+
+        :param x: Input with shape as expected by the model.
+        :type x: `Tensor`
+        :param y: Target values (class labels) one-hot-encoded of shape (nb_samples, nb_classes) or indices of shape
+                  (nb_samples,).
+        :type y: `Tensor`
+        :return: Gradients of the same shape as `x`.
+        :rtype: `Tensor`
+        """
+        import torch
+        from torch.autograd import Variable
+
+        # Check label shape
+        if self._reduce_labels:
+            y = torch.argmax(y, dim=1)
+
+        # Convert the inputs to Variable
+        x = Variable(x, requires_grad=True)
+
+        # Compute the gradient and return
+        model_outputs = self._model(x)
```

Comment:
Should we call self._model.train() or self._model.eval() to make sure the model is in the right mode for gradient calculation?

Reply:
No, we already do that in the set_learning_phase(self, train) of the classifier.